src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -659,7 +660,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
                 {
                     countOfExplicitNames++;
 
-                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.UnderlyingMemberName(i))
+                    if (TupleTypeSymbol.IsMemberNameReserved(name) && name != TupleTypeSymbol.TupleMemberName(i))
```

Comment:
Ok, discussed this further with Aleksey. Because of unavoidable shadowing due to wrapping (which would be very confusing), it is better to not allow "Rest" in any position.